modules/pricer/src/main/java/com/opengamma/strata/pricer/rate/bond/DiscountingFixedCouponBondTradePricer.java

```diff
@@ -0,0 +1,225 @@
+   * The result is expressed using the payment currency of the bond.
+   * <p>
+   * The z-spread is a parallel shift applied to continuously compounded rates or periodic
+   * compounded rates of the discounting curve.
+   * <p>
+   * Coupon payments of the underlying product are considered based on the settlement date of the trade. 
+   * 
+   * @param trade  the trade to price
+   * @param provider  the rates provider
+   * @param zSpread  the z-spread
+   * @param periodic  if true, the spread is added to periodic compounded rates,
+   *  if false, the spread is added to continuously compounded rates
+   * @param periodsPerYear  the number of periods per year
+   * @return the present value of the fixed coupon bond trade
+   */
+  public CurrencyAmount presentValueWithZSpread(
+      FixedCouponBondTrade trade,
+      LegalEntityDiscountingProvider provider,
+      double zSpread,
+      boolean periodic,
+      int periodsPerYear) {
+    LocalDate settlementDate = trade.getTradeInfo().getSettlementDate().get();
+    CurrencyAmount pvProduct = productPricer.presentValueWithZSpread(
+        trade.getProduct(), provider, zSpread, periodic, periodsPerYear, settlementDate);
+    return presentValueFromProductPresentValue(trade, provider, pvProduct);
+  }
+
+  private CurrencyAmount presentValueFromProductPresentValue(
+      FixedCouponBondTrade trade,
+      LegalEntityDiscountingProvider provider,
+      CurrencyAmount productPresentValue) {
+    CurrencyAmount pvProduct = productPresentValue.multipliedBy(trade.getQuantity());
+    CurrencyAmount pvPayment = presentValuePayment(trade, provider);
+    return pvProduct.plus(pvPayment);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the present value of the fixed coupon bond trade from the clean price of the underlying product.
+   * <p>
+   * The present value of the trade is the value on the valuation date.
+   * The result is expressed using the payment currency of the bond.
+   * 
+   * @param trade  the trade to price
+   * @param provider  the rates provider
+   * @param cleanPrice  the clean price.
+   * @return the present value of the fixed coupon bond trade
+   */
+  public CurrencyAmount presentValueFromCleanPrice(
+      FixedCouponBondTrade trade,
+      LegalEntityDiscountingProvider provider,
+      double cleanPrice) {
+    Security<FixedCouponBond> security = trade.getSecurity();
+    FixedCouponBond product = security.getProduct();
+    LocalDate settlementDate = trade.getTradeInfo().getSettlementDate().get();
+    StandardId securityId = security.getStandardId();
+    StandardId legalEntityId = product.getLegalEntityId();
+    double df = provider.repoCurveDiscountFactors(
+        securityId, legalEntityId, product.getCurrency()).discountFactor(settlementDate);
+    double pvPrice = (cleanPrice * product.getNotional() + productPricer.accruedInterest(product, settlementDate)) * df;
```

Comment:
The coupons received in the actual trade but not in the standard settlement (or the opposite) should be removed/added.
The PV of the settlement amount (pvPayment in other methods) should be added.